### DISCUSSION TITLE
how to use g2plot when use vitest

### DISCUSSION THREAD

**Discussion by luomaochen** [vitest-dev/vitest#3692] [Q&A]
In vue3 using g2plot development run packaged version of all normal, recently in the supplemental unit test, found vitest run with chart drawing page component has been because of the introduction of g2plot and error, the error is required() of ES Module to remove the introduction of the chart and drawing single test then normal operation

Error: require() of ES Module D:\projects\node_modules\d3-interpolate\src\index.js from D:\projects\node_modules@antv\g-base\lib\animate\timeline.js not supported.
Instead change the require of index.js in D:\projects\node_modules@antv\g-base\lib\animate\timeline.js to a dynamic import() which is available in all CommonJS modules.
❯ Object. node_modules/@antv/g-base/lib/animate/timeline.js:5:24
❯ Object. node_modules/@antv/g-base/lib/abstract/canvas.js:7:18
❯ Object. node_modules/@antv/g-base/lib/index.js:17:16
❯ Object. node_modules/@antv/g2/lib/dependents.js:6:16
❯ Object. node_modules/@antv/g2/lib/util/scale.js:7:20
❯ Object. node_modules/@antv/g2/lib/util/axis.js:6:15
❯ Object. node_modules/@antv/g2/lib/facet/facet.js:7:14
❯ Object. node_modules/@antv/g2/lib/facet/index.js:5:15
❯ Object. node_modules/@antv/g2/lib/chart/view.js:8:15
❯ Object. node_modules/@antv/g2/lib/chart/chart.js:8:38
❯ Object. node_modules/@antv/g2/lib/chart/index.js:7:15
❯ Object. node_modules/@antv/g2/lib/core.js:8:15
❯ Object. node_modules/@antv/g2/lib/index.js:6:14
❯ async D:/projects/dbadvisor-web/node_modules/@antv/g2plot/esm/index.js:1:31

how to add config to deal with the error  please help

**Comment by macaframaretard** [comment]
@sheremet-va Hi, can you help with this question?
